### PR TITLE
Fix issue w/ vars and enums, make sure benchmark queries work, update documentation

### PR DESCRIPTION
**Pull request by hlship**
This started as just updating the documentation, and along the way, I hit a bug with variables assigned to enum fields, which I've fixed.

### CODE REVIEW

File: src/com/walmartlabs/lacinia/parser.clj [hunk 1]
```diff
@@ -491,33 +491,43 @@
...
+    :let [nested-type (:type argument-type)
+          kind (:kind argument-type)]
 
-      list-type?
-      (if (not (sequential? result))
-        (throw-exception (format "Variable %s doesn't contain the correct number of (nested) lists."
-                                 (q arg-value))
-                         {:variable-name arg-value})
-        (mapv #(process-result schema % nested-type arg-value) result))
+         ;; we can only hit this if we iterate over list members
```
Comment (bcarrell) [inline src/com/walmartlabs/lacinia/parser.clj]: It seems like indentation gets screwed up from here down.

Comment (hlship) [inline src/com/walmartlabs/lacinia/parser.clj]: Looks like Cursive "forgot" how to indent cond-let.

File: src/com/walmartlabs/lacinia/parser.clj [hunk 2]
```diff
@@ -491,33 +491,43 @@
...
+    :let [nested-type (:type argument-type)
+          kind (:kind argument-type)]
+
+    ;; we can only hit this if we iterate over list members
+    (and (nil? result) (= :non-null kind))
+    (throw-exception (format "Variable %s contains null members but supplies the value for a list that can't have any null members."
+                             (q arg-value))
+                     {:variable-name arg-value})
+
+    (= :list (:kind argument-type))
```
Comment (annapawlicka) [inline src/com/walmartlabs/lacinia/parser.clj]: `(= :list kind)`

File: src/com/walmartlabs/lacinia/parser.clj [hunk 3]
```diff
@@ -491,33 +491,43 @@
...
-      :else
-      result)))
+    (= category :scalar)
+    (process-literal-argument schema {:type argument-type} [:scalar result])
+
+    ;; enums have to be handled carefully because they are likely strings in
+    ;; the variable map.
+
+    (= category :enum)
+    (process-literal-argument schema {:type argument-type} [:enum (as-keyword result)])
```
Comment (annapawlicka) [inline src/com/walmartlabs/lacinia/parser.clj]: Could we add a test for this scenario so that we avoid regression in the future?

**Comment by hlship** [comment]
Ok, back in your court.

**Pull request merged**